src/Export/ProductDebugService.php

```diff
@@ -0,0 +1,204 @@
+
+        /** @var ProductEntity $item */
+        $this->product = $this->fetchProduct();
+
+        if (!$this->product) {
+            $this->exportErrors->addGeneralError(
+                sprintf('Product or variant with ID %s does not exist.', $this->productId)
+            );
+
+            return new JsonResponse(
+                $this->exportErrors->buildErrorResponse(),
+                422
+            );
+        }
+
+        $exportedMainProductId = $this->exportedMainVariantId();
+        $isExported = $this->isExported() && !$exportErrors->hasErrors();
+
+        if (!$isExported) {
+            $this->checkExportCriteria();
+        }
+
+        return new JsonResponse([
+            'export' => [
+                'productId' => $this->product->getId(),
+                'exportedMainProductId' => $exportedMainProductId,
+                'isExported' => $isExported,
+                'reasons' => array_merge($this->parseExportErrors(), $this->reasons)
+            ],
+            'debugLinks' => [
+                'exportUrl' => $this->buildExportUrl($exportedMainProductId),
+                'debugUrl' => $this->buildDebugUrl($exportedMainProductId)
+            ],
+            'data' => [
+                'isExportedMainVariant' => $exportedMainProductId === $this->product->getId(),
+                'product' => $this->product,
+                'siblings' => $this->product->getParentId() ? $this->getSiblings($this->product) : [],
+                'associations' => $this->buildCriteria()->getAssociations(),
+            ]
+        ]);
+    }
+
+    public function fetchProduct(?string $productId = null, ?bool $withVariantInformation = false): ?ProductEntity
+    {
+        $criteria = $this->buildCriteria($productId);
+
+        /** @var EntitySearchResult $entityResult */
+        $entityResult = $this->getContainer()->get('product.repository')->search(
+            $criteria,
+            $this->getSalesChannelContext()->getContext()
+        );
+
+        return $withVariantInformation
+            ? $this->buildProductsWithVariantInformation($entityResult)->first()
+            : $entityResult->first();
+    }
+
+    private function searchProduct(Criteria $criteria): ?ProductEntity
+    {
+        return $this->getContainer()->get('product.repository')->search(
+            $criteria,
+            $this->getSalesChannelContext()->getContext()
+        )->first();
+    }
+
+    private function buildCriteria(?string $productId = null, ?bool $withAssociations = true): Criteria
+    {
+        $criteria = new Criteria([$productId ?? $this->productId]);
+
+        if ($withAssociations) {
+            Utils::addProductAssociations($criteria);
+        }
+
+        return $criteria;
+    }
+
+    private function exportedMainVariantId(): string
+    {
+        $product = $this->fetchProduct($this->productId, true);
+
+        return $product->getId();
+    }
+
+    private function isExported(): bool
```

Comment:
There are many checks in this class, maybe it makes sense to split this up in multiple smaller logical components.